### PR TITLE
Adding devfile and 'Developer Workspace' button in readme for trying the project on Hosted Che / che.openshift.io

### PR DESCRIPTION
**Pull request by ibuziuk**
Adding devfile and 'Developer Workspace' button in readme for trying the project on Hosted Che / che.openshift.io

Try the PR on Hosted Che:

[Contribute](https://che.openshift.io/f?url=https://github.com/ibuziuk/jbang/tree/devfile)

jbang

image

### CODE REVIEW

File: devfile.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+apiVersion: 1.0.0
+metadata:
+  name: jbang
+projects:
+  -
+    name: jbang
+    source:
+      type: git
+      location: "https://github.com/maxandersen/jbang.git"
```
Comment (maxandersen) [inline devfile.yaml]: this means forks wont use their own repo for edits - can we avoid that ?

File: devfile.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    id: redhat/java/latest
+  -
+    type: dockerimage
+    alias: gradle
+    image: quay.io/eclipse/che-java11-gradle:nightly
+    env:
+      - name: GRADLE_USER_HOME
+        value: /home/gradle/.gradle
+      - name: JAVA_OPTS
+        value: "-XX:MaxRAMPercentage=50 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
```
Comment (maxandersen) [inline devfile.yaml]: how can I avoid having to specify this in every devfile ? I don't want to maintain this...

File: devfile.yaml (hunk 1 above)
Comment (ibuziuk) [reply devfile.yaml]: I suppose the fork low you describe is not supported
@skabashnyuk  could you please confirm?
But in any case having a "Try it on Hosted Che" button for the original project is the step forward IMO.

File: devfile.yaml (hunk 2 above)
Comment (ibuziuk) [reply devfile.yaml]: I do not think it is possible atm - one need to explicitly specify dedicated env vars

File: devfile.yaml (hunk 1 above)
Comment (ibuziuk) [reply devfile.yaml]: fixed. Starting from 7.8.0 upstream version it is possible not to specify the project

Comment (maxandersen) [reply devfile.yaml]: awesome! do you know when this is available on che.openshift.io ?

Comment (ibuziuk) [reply devfile.yaml]: @maxandersen it is already available, you can try via the button in the PR description

**Comment by maxandersen** [comment]
sorry for the delay on trying this out (been bandwidth challenged the last month :)

Here is what I get when I click the button in this PR - a 401 error:

**Comment by maxandersen** [comment]
on second click it tells me my account needs activating.

I click activate and now its looking like its booting che.

so something in the auth check flow isn't right as it shouldn't have let me in the first time ;)

**Comment by maxandersen** [comment]
merged this.

**Pull request closed without merging**